lib/jsonapi/utils/string.ex

```diff
@@ -123,13 +175,20 @@ defmodule JSONAPI.Utils.String do
   defp normalized_underscore_to_dash_config(value) when is_nil(value), do: value
 
   @doc """
-  The configured transformation for the API's fields. JSON:API v1 recommends
-  using dashed fields (e.g. "good-dog", versus "good_dog").
+  The configured transformation for the API's fields. JSON:API v1.1 recommends
+  using camlized fields (e.g. "goodDog", versus "good_dog").  However, we don't hold a strong
+  opinion, so feel free to customize it how you would like (e.g. "good-dog", versus "good_dog").
 
-  This library currently supports dashed and underscored fields.
+  This library currently supports camelized, dashed and underscored fields.
 
   ## Configuration examples
 
+  camelCase fields:
+
+  ```
+  config :jsonapi, field_transformation: :camelize
```

Comment:
Ought this to be the default going forward, or do you think that explicit configuration is more ideal?

Reply:
Kind of goes back to this discussion but I agree it probably should be the default going forward.  Other libraries like ja_serializer also assume the default according to the jsonapi spec.  I'll wait for @doomspork's thoughts as well.

https://github.com/jeregrine/jsonapi/pull/149#discussion_r244522064